galter_subjects_utils/adapter.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021-2024 Northwestern University.
+#
+# galter-subjects-utils is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Generic download functionality."""
+
+from .types_internal import Subject
+
+
+def converted_to_subjects(rdm_iter, prefix):
+    """Adapts an RDMSubject ("Converted") iterable into a Subject iterable.
+
+    :param rdm_iter: Iterable[RDMSubject]
+
+    RDMSubject:
+
+    {
+        "id": "<prefix>...",
+        "scheme": "...",
+        "subject": "..."
+    }
+
+    :return: Iterable[Subject]
+
+    Subject:
+
+        id (minus the <prefix> part)
+        label
+    """
+    len_prefix = len(prefix)
+    return (
+        Subject(
+            id=e["id"][len_prefix:] if e["id"].startswith(prefix) else e["id"],
```

Comment:
Ah, that suggested change would have been much easier to read for me. But I defer, and you are right to be on the safer side by doing a check. It was indeed my misunderstanding of python; I have forgotten what the colon, `:`, did in `[len_prefix:]` 🤦 

I am starting to come around to the way Python's if-expressions work, and like the readability of it.

Reply:
Actually, another question on the use of `len_prefix` here. Why not match the prefix exactly, instead of it's length? Wouldn't that leave open a possibility of a differently worded prefix, but same length?

Reply:
(for the nevermind: we're doing the startswith() check so we are indeed good)

https://imgs.xkcd.com/comics/wisdom_of_the_ancients.png